src/python/gudhi/simplex_tree.pyx

```diff
@@ -627,11 +627,18 @@ cdef class SimplexTree:
             maximal dimension in the complex is computed. If false, it is
             ignored. Default is false.
         :type persistence_dim_max: bool
+        :param output_type: Format of the output. 'old' for the legacy list of (dim,(birth,death)),
+            'array by dimension' for a list of nx2 numpy arrays (one per dimension).
+        :type output_type: str
         :returns: The persistence of the simplicial complex.
         :rtype:  list of pairs(dimension, pair(birth, death))
         """
         self.compute_persistence(homology_coeff_field, min_persistence, persistence_dim_max)
-        return self.pcohptr.get_persistence()
+        if output_type == 'array by dimension':
```

Comment:
"array by dimension" seems a bit long to me, is "arrays" ok ? and 'old' isn't very telling :  "tuples" ?
Also, should we raise a [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) to change the default to the array one afterward ?